# game informantion and robot state
uint16 remain_hp
uint8 armor_type
uint8 hurt_type
uint8 bullet_freq
float64 bullet_speed

int16 remain_bullet
int16 sent_bullet

float64 x_speed
float64 y_speed
float64 gimbal_pitch
float64 gimbal_yaw